Sources/CvUnit.cpp

```diff
@@ -30860,51 +30807,40 @@ bool CvUnit::canAirBomb1(const CvPlot* pPlot) const
 
 bool CvUnit::canAirBomb1At(const CvPlot* pPlot, int iX, int iY) const
 {
-	CvCity* pCity;
-	CvPlot* pTargetPlot;
-
 	if (!canAirBomb1(pPlot))
 	{
 		return false;
 	}
 
-	pTargetPlot = GC.getMapINLINE().plotINLINE(iX, iY);
-
-	if (plotDistance(pPlot->getX_INLINE(), pPlot->getY_INLINE(), pTargetPlot->getX_INLINE(), pTargetPlot->getY_INLINE()) > airRange())
+	const CvPlot* pTargetPlot = GC.getMap().plot(iX, iY);
+	if (plotDistance(pPlot->getX(), pPlot->getY(), pTargetPlot->getX(), pTargetPlot->getY()) > airRange())
 	{
 		return false;
 	}
 
-	if (pTargetPlot->isOwned())
+	if (pTargetPlot->isOwned() && !atWar(pTargetPlot->getTeam(), getTeam()))
 	{
-		if (!atWar(pTargetPlot->getTeam(), getTeam()))
-		{
-			return false;
-		}
+		return false;
 	}
 
-	pCity = pTargetPlot->getPlotCity();
-
+	const CvCity* pCity = pTargetPlot->getPlotCity();
 	if (pCity != NULL)
 	{
-		if (!(pCity->isBombardable(this)))
+		if (!pCity->isBombardable(this))
```

Comment:
This statement can also be moved into the outer if: the right hand side of an && statement is only executed if the left side was true, so you can use it to check for null before using a value in the same statement.

Reply:
if (pCity != NULL)
	{
		if (!pCity->isBombardable(this))
		{
			return false;
		}
	}
	else
	{

That won't mess with the else?

Reply:
Yeah you are right I didn't read it properly!